contract/src/contractStarter.js

```diff
@@ -93,25 +155,47 @@ export const start = (zcf, _privateArgs, _baggage) => {
   const makeStartInvitation = async opts => {
     mustMatch(opts, StartOptionsShape);
 
+    const Fee = sum([
+      prices.startInstance,
+      ...('installation' in opts ? [] : [prices.installBundleID]),
+      ...keys(opts.permit || {}).map(k => prices[k]),
+    ]);
+
     /** @param {ZCFSeat} seat */
     const handleStart = async seat => {
+      atomicRearrange(zcf, harden([[seat, fees, { Fee }]]));
       const installation = await ('installation' in opts
         ? opts.installation
         : E(zoe).installBundleID(opts.bundleID));
 
       const { issuerKeywordRecord, customTerms, privateArgs, instanceLabel } =
         opts;
+      const { storageNode: nodePermit, ...permit } = opts.permit || {};
+      const powers = fromEntries(
+        keys(permit || {}).map(k => [k, limitedPowers[k]]),
+      );
       /** @type {StartedInstanceKit<SF>} */
       const it = await E(zoe).startInstance(
         installation,
         issuerKeywordRecord,
         customTerms,
-        privateArgs,
+        { ...privateArgs, ...powers },
         instanceLabel,
       );
       // WARNING: adminFacet is dropped
       const { instance, creatorFacet } = it;
 
+      const itsTerms = await E(zoe).getTerms(instance);
+      const itsId = await E(board).getId(instance);
+      const itsNode = await E(storageNode).makeChildNode(itsId);
+      await publish(itsNode, itsTerms, pubMarshaller);
+
+      if (nodePermit) {
+        const itsStorage = await E(itsNode).makeChildNode('info');
+        // @ts-expect-error nodePermit implies this method
+        await E(creatorFacet).initStorageNode(itsStorage);
```

Comment:
`resolveStorageNode` is perhaps a better name, since the started contract likely has a pending promise for the storageNode

Reply:
no, `init` is fine